### ISSUE TITLE
Updating HDFStore in place

### ISSUE THREAD

**Issue by rockg** [pandas-dev/pandas#6857]
Currently I use the HDF5 interface to store timeseries and it works great for selection.  However, the only way I see to update them is to select the existing stored timeseries and then merge the update with the existing data.  This can obviously be expensive if the existing series is large.  Is it possible to update a store by just passing in the new data?  If so, is there an example somewhere?

**Comment by spitz-dan-l** [comment]
Hello, I'm wondering if there's been any progress on adding .modify() to HDFStore. It'd be such a boon for me. Thanks!

**Comment by rockg** [comment]
Unfortunately there hasn't.  I agree that it would be great to add.  I have something started, but not even worth pushing somewhere at this point.

**Comment by jmerkow** [comment]
I would be interested in this too.  But this seems dead.
Does anyone have a good work-around?  Right now I believe you would need to delete the row with store.remove, the append it back with append.  Or re-write the whole table (in which case you probably would just write a new file).
Is this feature available on any of the other IO options (possibly sql)?

**Comment by jreback** [comment]
see my comment above, you can implement the ``modify`` methods. Note this is not very efficient, but it will work. An HDF5 store works best when you append only, deleting/modify requires repacking the files (well it doesn't require it), but it is much more efficient that way.

**Comment by jmerkow** [comment]
Yes I saw that. It seems that HDF5 would not work well for large data flows where the data is modified frequently even if someone were to write store.modify.
Do any of the other IO options already have methods to modify the table? Would it be more appropriate to add this feature to SQL IO?

P.S. I can take this to the mailing list if you prefer.

**Comment by jreback** [comment]
no, this is specifically for HDF5. You generally don't want to modify data, you simply append, that's how its designed and makes it performant. You can already do this with SQL if you really want.

**Comment by jmerkow** [comment]
> You can already do this with SQL if you really want.
Can you point me to this?

**Comment by jreback** [comment]
http://pandas.pydata.org/pandas-docs/stable/io.html#io-sql

you would simply do some kind of an update query.

**Comment by jbrockmendel** [comment]
pandas support for HDF is in bugfix-only mode, closing as no-action.
